couchdb31.spec

```diff
@@ -0,0 +1,71 @@
+### RPM external couchdb31 3.1.1
+Source0: https://downloads.apache.org/couchdb/source/%{realversion}/apache-couchdb-%{realversion}.tar.gz
+Source1: couchdb31_cms_auth.erl
+Patch0: couchdb31-ssl-cert-patch
+Patch1: couchdb31-fix-rep-streaming
+
+# Patch explanation
+#   couchdb31-ssl-client-cert:
+#     Passes cacert_file too for the ssl context in the replicator
+#     code. Otherwise, it cannot use proxy certificates.
+#   couchdb31-fix-rep-streaming:
+#     Fixes or at least avoids the obscure bug of couch replicator
+#     client where it does not process chunked requests correctly
+#     when replicating from (but not to) a remote couch behind
+#     a SSL (Apache) proxy. On this very specific case (which happens only
+#     when replicating big documents), it seems couch is not reading
+#     the very last received data in the SSL connection buffer.
+#     We tell ibrowse to handle the (pace of) reading of incoming data buffer
+#     instead of couch. Not clear if the bug is in couch or in ibrowse.
+
+# Although there is no technical software dependency,
+# couchapp was included because all CMS applications will need it.
+Requires: curl spidermonkey icu4c erlang22 py3-cmscouchapp
+BuildRequires: autotools
+
+%prep
+%setup -n apache-couchdb-%{realversion}
+%patch0 -p0
+%patch1 -p0
+#mv %_sourcedir/couchdb31_cms_auth.erl %_sourcedir/couch_cms_auth.erl
+#ls -l %_builddir/apache-couchdb-%realversion/src/couch
+cp %_sourcedir/couchdb31_cms_auth.erl %_builddir/apache-couchdb-%{realversion}/src/couch/src/couch_cms_auth.erl
+
+%build
+# apache 1.5.1 does not have option to specify --with-icu4c, instead
+# they used --with-win32-icu-binaries which mostly the same
+export CURL_ROOT SPIDERMONKEY_ROOT ICU4C_ROOT ERLANG_ROOT AUTOTOOLS_ROOT
```

Comment:
@amaltaro , please use ERLANG22_ROOT every where in this spec as you depend on `erlang22`

Reply:
Thanks for pointing this out, Shahzad. I have it fixed in the second commit.